### DISCUSSION TITLE
[Survey] Developer Experience

### DISCUSSION THREAD

**Discussion by dmathieu** [open-telemetry/opentelemetry-ruby#1726] [Ideas]
Hello all (@open-telemetry/ruby-approvers ) 👋🏽 

The DevEx SIG is running a survey with other SIGs, to collect feedback from the approvers and maintainers of each programming language.
We are initially focusing on the topics below, so we'd love to have your insights on it.

After we collect everyone's opinion, I'll sum this up in the DevEx repo: https://github.com/open-telemetry/sig-developer-experience/issues/14

Collecting this info would help us find spots to improve throughout the whole OTel project.

Feel free to link issues/discussions/docs and so on.

### Survey

1. Differences from upstream spec:
    - discrepancies - why?
    - extra convenience
2. Most requested features
    - API
    - SDK
    - Instrumentations/semantic conventions
3. Things to consider from other telemetry projects

**Comment by kaylareopelle** [comment]
Hi @dmathieu! I haven't forgotten about this, and I'm sorry for taking so long to get back to you. Thanks for your patience and for your willingness to bring our response back to the DevEx SIG.

These are great questions that I'd love to have clearly documented answers for. However, I don't have the answers readily available and it'll take some time to gather the information. 

For now, [I'm putting my findings in this GoogleDoc](https://docs.google.com/document/d/19LVmv6GKFhGcZnZly-mrkbgi-vQxyrx1pj4kYvjGX8s/edit?usp=sharing).

For question 1, are you looking for areas both where we don't meet the spec because a feature is missing as well as intentional deviations from the spec?

Also, could you provide some more detail about what you're hoping to learn from question 3? That one confuses me a bit.

**Reply by dmathieu** [reply]
This Google Doc looks like an awesome starting point. Thank you! 🌷

> For question 1, are you looking for areas both where we don't meet the spec because a feature is missing as well as intentional deviations from the spec?

Both, (the questions are in fact a starting point, but we're happy to hear more open ended opinions).
The intent there is to uncover specification pain of lacking points that would need to be fixed, or clarified.

> Things to consider from other telemetry projects

This would be things that other telemetry projects (especially written in ruby) would be doing, and that we don't (yet?).
